[basicprofiles] Add Inactivity profile (#18301)
openhab/openhab-addons: bundles/org.openhab.transform.basicprofiles/README.md

  channel="mybinding:mything:mychannel" [ profile="basic-profiles:state-filter", conditions=">= MinimumPowerLimit", "< MaximumPowerLimit" ]
}
```

## Inactivity Profile

This profile sets the state of the item to `ON` (resp. `OFF`) if the binding has not provided any new data values within a given timeout period.
The purpose is to indicate an alarm condition if the binding is no longer providing values for the given channel.

### Inactivity Profile Configuration

| Configuration Parameter | Type    | Description                                                                                                                                                             |
| ----------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `timeout`               | text    | The time out after which the profile will set the state of the item if the binding provides no updates. The value is in `QuantityType` format e.g. `1 h`, `60 s`, `2 d` |

Author: > if there's a choice between using QuantityType to parse the string and Duration to parse the string

I decided to do both :) .. 

```
 private Duration parseDuration(String timeOrDuration) throws IllegalArgumentException {
        if (timeOrDuration.isBlank()) {
            return DEFAULT_TIMEOUT;
        }
        try {
            if (QuantityType.valueOf(timeOrDuration) instanceof QuantityType<?> time
                    && time.toUnit(MetricPrefix.MILLI(Units.SECOND)) instanceof QuantityType<?> msec) {
                return Duration.ofMillis(msec.longValue());
            }
        } catch (IllegalArgumentException e) {
            // fall through
        }
        // TODO refactor when #4868 is merged: return DurationUtils.parse(timeOrDuration);
        throw new IllegalArgumentException("invalid time or duration");
    }
```

  channel="mybinding:mything:mychannel" [ profile="basic-profiles:state-filter", conditions=">= MinimumPowerLimit", "< MaximumPowerLimit" ]
}
```

## Inactivity Profile

This profile sets the state of the item to `ON` if the binding has not provided any new data values within a given timeout period.
The purpose is to indicate an alarm condition if the binding is no longer providing values for the given channel.

### Inactivity Profile Configuration

| Configuration Parameter | Type    | Description                                                                                                                                                                                                                                                                      |
|-------------------------|---------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `timeout`               | text    | The time out after which the profile will set the state of the item if the binding provides no updates. The default is 1h. It uses the same format as the [expire parameter]( https://www.openhab.org/docs/configuration/items)                                                  |

Reviewer: ```suggestion
| `timeout`               | text    | Duration after which the item's state is updated if no data is received from the binding. The default is 1h. The format is the same as used in the [expire parameter](https://www.openhab.org/docs/configuration/items). |
```

> | `inverted`              | boolean | Optional. When `false` (the default), the item's state is initially set to `OFF`, and when the binding provided no data within the timeout period, the item's state is updated to `ON`. When `inverted` is set to `true`, the initial state and the timeout states are inverted. |

Reviewer: ```suggestion
| `inverted`              | boolean | Optional. If false (default), the item's state is initially OFF, and changes to ON after the timeout. If true, the behavior is reversed: the initial state is ON, and it changes to OFF after the timeout. |
```
